snappiershot/compare.py

```diff
@@ -43,8 +45,16 @@ def _compare(self, value: Any, expected: Any, *, operations: List[Callable]) ->
               Tracks the operations that need to be applied to self.value and self.expected
                 to obtain value and expected, respectively. Used for logging differences.
         """
-        # Check the types of both objects.
-        if type(value) != type(expected):
+        # Special check for units since it's possible the types won't match even if they are equal objects
+        if isinstance(expected, Unit) or isinstance(value, Unit):
+            return self._compare_units(value, expected, operations=operations)
```

Comment:
sometimes `expected` is a string that says `meter` while `value` is a `Unit` that says `<Unit('meter')>'. the reason for this is due to the recursion. recursion while serializing stops once it reaches an object that can be turned into a dictionary either via `__slots__`, '__dict__', or a `to_dict' method. based on the one used, a `Unit` may get serialized into a plain old dictionary with a key of 'unit' and value of `meter` or a key of  `__snappiershot_unit__' with a value of `meter`. so, sometimes units get deserialized into an actual `Unit` and othertimes it's just a string. it was blowing my mind trying to fix this so i just left the code as an `or` statement

Reply:
Yeah this is a tricky one and I guess we have to do our best to ensure it works as expected. Would be good to have more opinions on this

Reply:
I think it's ok to allow the OR case in which either the reference (snapshot) or test might be a Unit type since the deserialized check outputs the string rep of that Unit. The pint.Unit type instance check passes where expected/value could be the aforementioned Unit type cases, so I think this approach covers the most common use cases of pint.Unit.

Reply:
I will make a ticket to look into this more. If it covers the common use cases then I think we are good to leave it in and be on the look out for problems.